lib/BackbeatConsumer.js

```diff
@@ -224,18 +226,14 @@ class BackbeatConsumer extends EventEmitter {
 
         this._processingQueue = async.queue(
             this._queueProcessor, this._concurrency);
-        // when the task queue is empty, commit offset for all
-        // consumed partitions and try consuming new messages right away
+
+        // bookkeeping tasks when queue is empty
         this._processingQueue.drain = () => {
             // when auto-commit is set we don't have to commit
             // explicitly, so let's not do it for performance reasons
             if (!this._autoCommit) {
                 this._consumer.commit();
             }
-            this.emit('consumed', this._messagesConsumed);
```

Comment:
If autocommit is not enabled (which is the case for crr) and now that we are switching to flowing mode, when there are a lot of messages, the queue would never be empty (correct?) So the manual commit would probably happen after n messages and n could be 10/100/200?

Reply:
We are actually not switching to flowing mode, we are still running in non-flowing mode by consuming a specific amount of messages (https://github.com/scality/backbeat/pull/399/files#diff-e2120238e136b010d97978588107a8cbR278).

The `'consumed'` event is right now only used in tests AFAICS, and I moved it from this place to https://github.com/scality/backbeat/pull/399/files#diff-e2120238e136b010d97978588107a8cbR278, where each ending task has an opportunity to emit this event.

Reply:
I realize this deserves a more specific answer.

Even though we're still in non-flowing mode, you're still right in saying the queue may never get empty, this until we stop adding new messages to the queue at least. This behavior has the same kind of validity that it used to, but it may have become less useful since there's no guarantee that it will be called timely.

To overcome this, the caller either has to use auto-commit, or we should add a `commit()` wrapper that the user is responsible for calling when he wishes to commit the current offset. For our use-case it seems reasonable to use auto-commit for replication. It may be also worth enabling it by default, like the kafka client does.

Reply:
This has been solved by enabling auto-commit on the kafka consumer.